include/secp256k1_musig.h

```diff
@@ -0,0 +1,428 @@
+ *  `musig_partial_signature_serialize` and `musig_partial_signature_parse`
+ *  functions.
+ */
+typedef struct {
+    unsigned char data[32];
+} secp256k1_musig_partial_signature;
+
+/** Computes a combined public key and the hash of the given public keys
+ *
+ *  Returns: 1 when the public keys were successfully combined, 0 otherwise
+ *  Args:        ctx: pointer to a context object initialized for verification
+ *                    (cannot be NULL)
+ *           scratch: scratch space used to compute the combined pubkey by
+ *                    multiexponentiation. If NULL, an inefficient algorithm is used.
+ *  Out: combined_pk: the MuSig-combined public key (cannot be NULL)
+ *         pk_hash32: if non-NULL, filled with the 32-byte hash of all input public
+ *                    keys in order to be used in `musig_session_initialize`.
+ *   In:     pubkeys: input array of public keys to combine
+ *         n_pubkeys: length of pubkeys array
+ */
+SECP256K1_API int secp256k1_musig_pubkey_combine(
+    const secp256k1_context* ctx,
+    secp256k1_scratch_space *scratch,
+    secp256k1_pubkey *combined_pk,
+    unsigned char *pk_hash32,
+    const secp256k1_pubkey *pubkeys,
+    size_t n_pubkeys
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(5);
+
+/** Initializes a signing session for a signer
+ *
+ *  Returns: 1: session is successfully initialized
+ *           0: session could not be initialized: secret key or secret nonce overflow
+ *  Args:         ctx: pointer to a context object, initialized for signing (cannot
+ *                     be NULL)
+ *  Out:      session: the session structure to initialize (cannot be NULL)
+ *            signers: an array of signers' data to be initialized. Array length must
+ *                     equal to `n_signers` (cannot be NULL)
+ * nonce_commitment32: filled with a 32-byte commitment to the generated nonce
+ *                     (cannot be NULL)
+ *  In:  session_id32: a *unique* 32-byte ID to assign to this session (cannot be
+ *                     NULL). If a non-unique session_id32 was given then a partial
+ *                     signature will LEAK THE SECRET KEY.
+ *              msg32: the 32-byte message to be signed. Shouldn't be NULL unless you
+ *                     require sharing nonces before the message is known because it
+ *                     reduces nonce misuse resistance. If NULL, must be set with
+ *                     `musig_session_set_msg` before signing and verifying.
+ *        combined_pk: the combined public key of all signers (cannot be NULL)
+ *          pk_hash32: the 32-byte hash of the signers' individual keys (cannot be
+ *                     NULL)
+ *            pubkeys: input array of public keys. Array length must equal to
+ *                     `n_signers`
+ *          n_signers: length of signers and pubkeys array. Number of signers
+ *                     participating in the MuSig
```

Comment:
must be > 0 ? (Or say that pubkeys cannot be NULL)

Reply:
both. fixing